Add move operators to some classes

nyalldawson opened the pull request:
As identified as having a performance benefit by coverity scan

nyalldawson committed:
Add move operators to QgsRenderContext, QgsFeatureRequest

troopa81 reviewed (commented):
Shall we also not add move constructor so we respect [the rule of five](https://en.cppreference.com/w/cpp/language/rule_of_three.html) ?

nyalldawson committed:
Add move constructors too

nyalldawson committed:
Add move operators to QgsConditionalStyle

nyalldawson committed:
Add move operators to QgsMeshSpatialIndex

nyalldawson committed:
Add move operators to QgsCoordinateTransformContext

nyalldawson committed:
Add move operators to QgsRelation

nyalldawson committed:
Add move operators to QgsExpressionContextScope, QgsExpressionContext

nyalldawson committed:
Move operators for QgsDistanceArea

nyalldawson committed:
Move and copy operators for QgsTextDocument

nyalldawson committed:
Move operators for QgsDiagramLayerSettings

nyalldawson committed:
Move operators for QgsTiledSceneIndex

nyalldawson committed:
Move and copy operators for QgsMesh3DDataBlock

nyalldawson committed:
Move operators for QgsPolymorphicRelation

nyalldawson committed:
Move operators for QgsColorRampLegendNodeSettings

nyalldawson committed:
Remove unncessary copy constructor/assignment operator from QgsRelationContext

This class contains a single raw pointer only, the default implementations are fine

nyalldawson committed:
Move and copy operators for QgsMeshRendererSettings

nyalldawson committed:
Move operators for text format classes

nyalldawson committed:
Move operators for QgsLabelingEngineSettings

nyalldawson committed:
Move operators for QgsPlotData

nyalldawson committed:
Suppress unwanted clang-tidy warning

nyalldawson committed:
Fix all bugprone-unhandled-self-assignment errors

nirvn reviewed (approved):
Gosh I wish we had a continuous benchmark setup to see the kind of impact these optimizations potentially bring on.
